TicTacToeConsoleApp/Program.cs

```diff
@@ -0,0 +1,174 @@
+        private static void InitializeBoard()
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                for (int j = 0; j < 3; j++)
+                {
+                    board.Add(i + " " + j, " ");
+                }
+            }
+        }
+
+        private static void PrintWinner()
+        {
+            Console.WriteLine("Игрок " + currentPlayer + " победил!");
+        }
+
+        private static void PrintNonWinner()
+        {
+            Console.WriteLine("Игра окончена вничью!");
+        }
+
+        // Вывод игрового поля на экран
+        private static void PrintBoard()
+        {
+            Console.Clear();
+            Console.WriteLine("   0  1  2");
+            for (int i = 0; i < 3; i++)
+            {
+                Console.Write(i + " ");
+                for (int j = 0; j < 3; j++)
+                {
+                    Console.Write("[" + board[i + " " + j] + "]");
+                }
+                Console.WriteLine();
+            }
+            Console.WriteLine();
+        }
+
+        // Проверка на корректность хода
+        private static bool IsValidMove(string move)
+        {
+            if (!board.ContainsKey(move))
+            {
+                return false;
+            }
+            if (board[move] != " ")
+            {
+                return false;
+            }
+            return true;
+        }
+
+        // Проверка на победу
+        private static bool IsWinResult()
+        {
+            // Проверка по строкам и столбцам
+            for (int i = 0; i < 3; i++)
+            {
+                if (board[i + " 0"] == currentPlayer && board[i + " 1"] == currentPlayer && board[i + " 2"] == currentPlayer)
+                {
+                    return true;
+                }
+                if (board["0 " + i] == currentPlayer && board["1 " + i] == currentPlayer && board["2 " + i] == currentPlayer)
+                {
+                    return true;
+                }
+            }
+
+            // Проверка по диагоналям
+            if (board["0 0"] == currentPlayer && board["1 1"] == currentPlayer && board["2 2"] == currentPlayer)
+            {
+                return true;
+            }
+            if (board["0 2"] == currentPlayer && board["1 1"] == currentPlayer && board["2 0"] == currentPlayer)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        // Проверка на ничью
+        private static bool IsDrawResult()
+        {
+            foreach (KeyValuePair<string, string> cell in board)
+            {
+                if (cell.Value == " ")
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        // Смена текущего игрока
+        private static void SwitchPlayer()
+        {
+            currentPlayer = currentPlayer == "X"
+                ? "O"
+                : "X";
+        }
```

Comment:
rework 
through the use of an index.
You can make a formula where you will always have either 0 or 1

Reply:
currentIndex = ++currentIndex % _players.Count

\\
1.
current = 0
currentIndex = ++0 % 2 
currentIndex = 1 % 2 \\ 1

2.
current = 1
currentIndex = ++1 % 2 
currentIndex = 2 % 2 \\ 0